packages/driver/src/cypress/cy.js

```diff
@@ -258,6 +261,11 @@ const create = function (specWindow, Cypress, Cookies, state, config, log) {
 
       contentWindow.CSSStyleSheet.prototype.insertRule = _.wrap(insertRule, cssModificationSpy)
       contentWindow.CSSStyleSheet.prototype.deleteRule = _.wrap(deleteRule, cssModificationSpy)
+
+      // drop "fetch" polyfill that replaces it with XMLHttpRequest
+      // by using "eval" we force the polyfill to use XMLHttpRequest objects
+      // from the app iframe that we wrap for network stubbing
+      contentWindow.eval(`fetch=${fetchPolyfill}`)
```

Comment:
I think we can pretty easily avoid eval + toString and use the correct XMLHttpRequest object if we patch-package and take in the XMLHttpRequest (or content window) reference

Reply:
ughh, but that would be more complex to set up and maintain. Remember the has-binary2 disaster